packages/browser-data-collector/src/index.ts

```diff
@@ -41,7 +40,10 @@ export const stop = async ( id: string ): Promise< boolean > => {
 
 	inFlightReporters.delete( id );
 
-	await existingReport.stop();
+	// The report may be still starting, wait for it.
+	const startedReport = await existingReport;
+
+	const payload = await startedReport.stop();
```

Comment:
Having a method `stop()` that must be called before another method (`toJSON()`) is not a great API. Given that I can't imagine any use case for stopping a report and not wanting to serialize it to JSON straight away, I unified both methods.